File: action-type/13-template/java/Main.java
package main;

public class Main {

    public static void main(String[] args) {
    // 直接 new
    ConfigLoader<Object> jsonLoader = new JSONConfigLoader();
    ConfigLoader<Object> yamlLoader = new YamlConfigLoader();

    System.out.println("--- 加载 JSON 配置（new） ---");
    Object jsonConfig = jsonLoader.load("config.json");

    System.out.println("--- 加载 YAML 配置（new） ---");
    Object yamlConfig = yamlLoader.load("config.yaml");

    // 工厂方法
    System.out.println("--- 加载 JSON 配置（工厂） ---");
    ConfigLoader<Object> jsonLoader2 = AbstractConfigLoader.createLoader("config.json");
    Object jsonConfig2 = jsonLoader2.load("config.json");

    System.out.println("--- 加载 YAML 配置（工厂） ---");
    ConfigLoader<Object> yamlLoader2 = AbstractConfigLoader.createLoader("config.yaml");
    Object yamlConfig2 = yamlLoader2.load("config.yaml");
    }
    
}

File: action-type/13-template/go/main.go
package main

import "fmt"

func main() {
	fmt.Println("--- 加载 JSON 配置（new） ---")
	jsonLoader := &JSONConfigLoader{}
	jsonLoader.Load("config.json")

	fmt.Println("--- 加载 YAML 配置（new） ---")
	yamlLoader := &YamlConfigLoader{}
	yamlLoader.Load("config.yaml")

	fmt.Println("--- 加载 JSON 配置（工厂） ---")
	jsonLoader2 := CreateLoader("config.json")
	jsonLoader2.Load("config.json")

	fmt.Println("--- 加载 YAML 配置（工厂） ---")
	yamlLoader2 := CreateLoader("config.yaml")
	yamlLoader2.Load("config.yaml")
}
